src/wanderer.ts

```diff
@@ -200,15 +215,66 @@ function guzzlrAbandonQuest() {
 
   if (
     // consider abandoning
-    !location || // if mafia faled to track the location correctly
-    !canAdvOrUnlock(location) || // or the zone is marked as "generally cannot adv"
+    !location || // if mafia failed to track the location correctly
+    !canAdvOrUnlock(location, true) || // or the zone is marked as "generally cannot adv"
     (globalOptions.ascending && wandererTurnsAvailableToday(location) < remaningTurns) // or ascending and not enough turns to finish
   ) {
     print("Abandoning...");
     Guzzlr.abandon();
   }
 }
 
+// NOTE: This list is in Priority Order as well.
+const ALL_FISH = ["carp", "cod", "hatchetfish", "bass", "trout", "tuna"] as const;
+type FishTuple = typeof ALL_FISH;
+type Fish = FishTuple[number];
+
+function fishLocations(): { [fish in Fish]: Location | null } {
+  const locations: { [fish in Fish]: Location | null } = {
+    carp: null,
+    cod: null,
+    trout: null,
+    bass: null,
+    hatchetfish: null,
+    tuna: null,
+  };
+
+  if (have($item`Clan VIP Lounge key`)) {
+    const floundryPage = visitUrl("clan_viplounge.php?action=floundry");
+    for (const fish of ALL_FISH) {
+      const regexp = /<b>${fish}:<\/b> ([A-Za-z\- ]+)<br>/;
+      const match = regexp.exec(floundryPage);
+      if (match && match[1]) {
+        const fishLocation = toLocation(match[1]);
+        if (fishLocation !== $location`none`) {
+          locations[fish] = fishLocation;
+        }
+      }
+    }
+  }
+
+  return locations;
+}
+
+function fishTargets(): WandererTarget[] {
+  if (have($item`Clan VIP Lounge key`)) {
+    return withVIPClan<WandererTarget[]>(() => {
+      const locations = fishLocations();
+      let value = 8;
+      return ALL_FISH.map((fish) => {
+        value -= 1;
```

Comment:
I don't like changing the value of `value` in the map function--could you use something more like
```suggestion
      return ALL_FISH.map((fish, index) => {
        const value = 7 - index;
```
?